pkg/artifactory/resource_artifactory_remote_helm_repository.go

```diff
@@ -13,11 +13,31 @@ var helmRemoteSchema = mergeSchema(baseRemoteSchema, map[string]*schema.Schema{
 		Description: "Base URL for the translation of chart source URLs in the index.yaml of virtual repos. " +
 			"Artifactory will only translate URLs matching the index.yamls hostname or URLs starting with this base url.",
 	},
+	"external_dependencies_enabled": {
+		Type:        schema.TypeBool,
+		Default:     false,
+		Optional:    true,
+		ForceNew:    true,
+		Description: "When set, external dependencies are rewritten.",
+	},
+	"external_dependencies_patterns": {
+		Type:     schema.TypeList,
+		Optional: true,
+		ForceNew: true,
+		Elem: &schema.Schema{
+			Type: schema.TypeString,
+		},
+		RequiredWith: []string{"external_dependencies_enabled"},
+		Description: "An Allow List of Ant-style path expressions that specify where external dependencies may be downloaded from. " +
+			"By default, this is set to ** which means that dependencies may be downloaded from any external source.",
+	},
 })
 
 type HelmRemoteRepo struct {
 	RemoteRepositoryBaseParams
-	HelmChartsBaseURL string `hcl:"helm_charts_base_url" json:"chartsBaseUrl,omitempty"`
+	HelmChartsBaseURL            string   `hcl:"helm_charts_base_url" json:"chartsBaseUrl,omitempty"`
+	ExternalDependenciesEnabled  bool     `hcl:"external_dependencies_enabled" json:"externalDependenciesEnabled,omitempty"`
```

Comment:
Should we continue using `omitempty` here for these new attributes? @chb0github My vote is no but would like to hear your take.

Reply:
`GoVirtualRepositoryParams` has `omitempty`, `DockerRemoteRepository` has not.

Reply:
Aha, if `omitempty` is removed from `external_dependencies_enabled`, it does not need `ForceNew`. I agree that it should be removed.

Reply:
Yeah, `omitempty` has been a constant/consistent troublemaker for us. Hence I'd like to not add this if at all possible. This should apply to `external_dependencies_patterns` as well.

Reply:
@tenstad - Our experience, over-and-over, has been that `omitempty` is generally bad. It allows the server to interpret its absence as it sees fit, and that behavior is inconsistent and opaque; varying from endpoint-to-endpoint as a given developer decided to handle it. 

Compounding the problem is what go considers `empty` - `0` for int is `empty`, but many times it's a perfectly valid value. Some with `""` blank strings. 

It's not a blanket ban, but using `omitempty` should be used on a case-by-case basis for specific needs. We currently have at least 3 tickets open relating to this

Please remove `omitempty` and put proper default values in those fields so it's clear what the expectations are

Reply:
We all agree, `omitempty` is removed :smile: